File: src/ast/map.zig
const SourceLoc = @import("../parser/token.zig").SourceLoc;

pub const MapDecl = struct {
    name: []const u8,
    map_type: MapType,
    key_type: Type,
    value_type: Type,
    max_entries: u32,
    loc: SourceLoc,
};

pub const MapType = enum {
    hash,
    array,
    ringbuf,
    lru_hash,
    prog_array,
    perf_event_array,
};

pub const Type = enum {
    u32,
    u64,
    i32,
    i64,
};

File: src/ast/map.rs

use crate::parser::SourceLoc;

#[derive(Debug, Clone)]
#[allow(unused)]
pub struct MapDecl {
    pub name: String,
    pub map_type: MapType,
    pub key_type: Type,
    pub value_type: Type,
    pub max_entries: u32,
    pub loc: SourceLoc,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
#[allow(unused)]
pub enum MapType {
    Hash,
    Array,
    Ringbuf,
    LruHash,
    ProgArray,
    PerfEventArray,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
pub enum Type {
    U32,
    U64,
    I32,
    I64,
}